libz-rs-sys/src/gz.rs

```diff
@@ -253,17 +277,18 @@ unsafe fn free_state(state: &mut GzState) {
 ///
 /// # Safety
 ///
-/// This function may be called at most once for any file handle.
+/// This function may be called at most once for any file handle. The caller is responsible
+/// for ensuring that the file handle either points to a GzState object or is null.
```

Comment:
In the main library we use this phrasing for what "valid" means. It's not just about being able to address the bytes, it also must e.g. by allocated by our allocator.

> /// * Either
> ///     - `file` is `NULL`
> ///     - `file` satisfies the requirements of `&mut *file` and was initialized with [`gzopen`] or similar

Reply:
I don't understand the part about "`file` satisfies the requirements of `&mut *file`." The requirements of `&mut *file` in this case aren't sufficient, since `file` is a pointer to a `gzFile_s` and we need the caller to pass a pointer to a specific subclass of `gzFile_s`, the `GzState`.

Reply:
right, maybe it's clearest to say that it must either be `NULL` or the output of `gzopen` (as far as I can tell `gzopen` is the only way to create a `gzFile`)? Given that `gzFile` is opaque, it is just never valid for a user to construct one of these on their own.

Reply:
Thanks, that makes sense. I'll add a comment saying the argument must have been provided by `gzopen`.